### DISCUSSION TITLE
[@types/lodash] omit doesn't remove any keys if nonexistent key specified

### DISCUSSION THREAD

**Discussion by erezarnon** [DefinitelyTyped/DefinitelyTyped#67503] [Issues with a @types package]
The below at the very least should error that `a` doesn't exist because it's been omitted. In fact, it should probably error before that, because we're trying to omit `b`, which doesn't exist.

```
_.omit({a: 2}, ['a', 'b']).a
```

[a comment by an automated account is not shown]

**Comment by aj-r** [comment]
That might be overly restrictive (and could break some people's existing
code) since typescript allows you to assign an object with more properties
than necessary to an interface type. So extra properties might exist at
runtime even if the compiler doesn't know about them.

**Comment by erezarnon** [comment]
I think what you're pointing out is that Typescript allows you to do things like:

```
let a = {a: 2};
const b = {a: 2, b: 3};
a = b;
```

That's true, but that's not what's happening here. Typescript doesn't allow you to do the below:

```
const a = {a: 2};
a.b;
```

**Comment by aj-r** [comment]
Ah yes, at the very least I agree that the `.a` part (in your original
example) should be an error. It might require some creative types... I'm
happy to review a PR if someone can figure it out.

**Comment by erezarnon** [comment]
The offending overload seems to be `omit<T extends object>(object: T | null | undefined, ...paths: Array<Many<PropertyName>>): PartialObject<T>;`. Why does that overload exist? It seems too relaxed whereas the other overloads cover the valid cases.
